packages/truffle-core/lib/testing/soliditytest.js

```diff
@@ -183,27 +183,25 @@ const SolidityTest = {
       "AssertIntArray",
       "AssertString",
       "AssertUint",
-      "AssertUintArray"
+      "AssertUintArray",
+      "DeployedAddresses",
     ];
 
-    const assertAbstractions = assertLibraries.map(name =>
+    const testAbstractions = testLibraries.map(name =>
       runner.config.resolver.require(`truffle/${name}.sol`)
     );
 
-    const DeployedAddresses = runner.config.resolver.require(
-      "truffle/DeployedAddresses.sol"
-    );
     SafeSend = runner.config.resolver.require(SafeSend);
 
-    for (const abstraction of assertAbstractions) {
-      deployer.deploy(abstraction);
+    for (const testLib of testAbstractions) {
+      deployer.deploy(testLib);
+      deployer.link(testLib, abstraction);
```

Comment:
Ah, so it looks like this line means that instead of your Solidity test contract only getting linked to the specific files it imports, this will now link to all these utils individually. Is this necessary? What happens if you remove this step?